automata/base/utils.py

```diff
@@ -10,17 +10,18 @@ def freezeValue(value):
     immutable one by recursively processing said structure and any of its
     members, freezing them as well
     """
-    if isinstance(value, list):
-        return tuple(freezeValue(element) for element in value)
+    if isinstance(value, (str, int)):
```

Comment:
I also added this check to avoid the 3 consecutive IFs on the observation that there should be as many primitive str/ints as non-primitive dict/set/list